### PR TITLE
Add null constants, `br` instruction, and clean up API

### PR DESCRIPTION
**Pull request by samarsha**
- Add `Constant.null(Type)`.
- Add `Builder.br(BasicBlock)`.
- Rename `constant_bytes` to `extract_bytes`.

### CODE REVIEW

File: pyqir/src/values.rs [hunk 1]
```diff
@@ -113,36 +114,33 @@ impl Value {
 ///
 /// :param Context context: The global context.
```
Comment (idavis) [inline pyqir/src/values.rs:115]: Please clarify the expected behavior based on the parameters.

File: pyqir/src/values.rs [hunk 2]
```diff
@@ -385,6 +409,40 @@ impl From<Linkage> for inkwell::module::Linkage {
...
+impl ConstantExpr {
+    /// Creates a `getelementptr` (GEP) constant expression.
+    ///
+    /// :param Value value: The aggregate value.
+    /// :param Sequence[Value] indices: The indices of the element.
+    /// :param bool inbounds: Whether to create an in-bounds GEP.
+    /// :returns: The GEP constant expression.
+    /// :rtype: ConstantExpr
+    #[staticmethod]
+    #[allow(clippy::needless_pass_by_value)]
```
Comment (idavis) [inline pyqir/src/values.rs:425]: Please add a test for inbounds == false

File: pyqir/tests/test_entry_points.py [hunk 3]
```diff
@@ -84,3 +84,19 @@ def test_invalid_module() -> None:
         verify_module(module)
         == "Basic Block in function 'main' does not have terminator!\nlabel %0\n"
     )
+
+
+def test_multiple_blocks() -> None:
```
Comment (idavis) [inline pyqir/tests/test_entry_points.py:89]: Please add tests for the matrix of options

**Review by idavis — CHANGES_REQUESTED** [review]

File: pyqir/src/values.rs (hunk 2 above)
Comment (samarsha) [reply pyqir/src/values.rs]: For some reason, LLVM doesn't seem to care if I use inbounds or not - it emits `getelementptr inbounds` no matter what... so I just removed the parameter.

**Review by idavis — APPROVED** [review]

**Pull request merged**